src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -490,11 +490,10 @@ private static String getImportedTypeCanonicalName(DetailAST importAst) {
         while (toVisit != null) {
             toVisit = getNextSubTreeNode(toVisit, importAst);
             if (toVisit != null && toVisit.getType() == TokenTypes.IDENT) {
-                canonicalNameBuilder.append(toVisit.getText());
-                final DetailAST nextSubTreeNode = getNextSubTreeNode(toVisit, importAst);
-                if (nextSubTreeNode.getType() != TokenTypes.SEMI) {
+                if (canonicalNameBuilder.length() > 0) {
                     canonicalNameBuilder.append('.');
                 }
+                canonicalNameBuilder.append(toVisit.getText());
```

Comment:
Actually, this method can be replaced with
`FullIdent.createFullIdentBelow(importAst).getText()`

Reply:
@pbludov This won't work.
For `import java.*` FullIdent will return `java.*` instead of `java` which is what this change requires. It must also return `java.Class` for `import java.Class`.
We can't add star checks to this method because we already ensure it is not a star import before calling this method. We probably wouldn't clear code coverage.

This section doesn't seem suitable for `FullIdent`.